light_sdk/src/reading/mod.rs

```diff
@@ -0,0 +1,679 @@
+        TendermintAddress::from_str(tendermint_addr)
+            .map_err(|e| Error::Other(e.to_string()))?,
+    )
+    .map_err(|e| Error::Other(e.to_string()))?;
+    let rt = Runtime::new().unwrap();
+    rt.block_on(rpc::query_native_token(&client))
+}
+
+/// Query the last committed block, if any.
+pub fn query_block(tendermint_addr: &str) -> Result<Option<LastBlock>, Error> {
+    let client = HttpClient::new(
+        TendermintAddress::from_str(tendermint_addr)
+            .map_err(|e| Error::Other(e.to_string()))?,
+    )
+    .map_err(|e| Error::Other(e.to_string()))?;
+    let rt = Runtime::new().unwrap();
+    rt.block_on(rpc::query_block(&client))
+}
+
+/// Query the results of the last committed block
+pub fn query_results(
+    tendermint_addr: &str,
+) -> Result<Vec<BlockResults>, Error> {
+    let client = HttpClient::new(
+        TendermintAddress::from_str(tendermint_addr)
+            .map_err(|e| Error::Other(e.to_string()))?,
+    )
+    .map_err(|e| Error::Other(e.to_string()))?;
+    let rt = Runtime::new().unwrap();
+    rt.block_on(rpc::query_results(&client))
+}
+
+pub mod tx {
+    use namada_sdk::events::Event;
+    use namada_sdk::rpc::{TxEventQuery, TxResponse};
+
+    use super::*;
+
+    /// Call the corresponding `tx_event_query` RPC method, to fetch
+    /// the current status of a transation.
+    pub fn query_tx_events(
+        tendermint_addr: &str,
+        tx_event_query: TxEventQuery<'_>,
+    ) -> std::result::Result<Option<Event>, Error> {
+        let client = HttpClient::new(
+            TendermintAddress::from_str(tendermint_addr)
+                .map_err(|e| Error::Other(e.to_string()))?,
+        )
+        .map_err(|e| Error::Other(e.to_string()))?;
+        let rt = Runtime::new().unwrap();
+        rt.block_on(
+            rpc::query_tx_events(&client, tx_event_query)
+        ) .map_err(|e| Error::Other(e.to_string()))
+    }
+
+    /// Dry run a transaction
+    pub fn dry_run_tx(
+        tendermint_addr: &str,
+        tx_bytes: Vec<u8>,
+    ) -> Result<namada_core::types::transaction::TxResult, Error> {
+        let client = HttpClient::new(
+            TendermintAddress::from_str(tendermint_addr)
+                .map_err(|e| Error::Other(e.to_string()))?,
+        )
+        .map_err(|e| Error::Other(e.to_string()))?;
+        let (data, height, prove) = (Some(tx_bytes), None, false);
+        let rt = Runtime::new().unwrap();
+        let result = rt
+            .block_on(RPC.shell().dry_run_tx(&client, data, height, prove))
+            .map_err(|err| {
+                Error::from(namada_sdk::error::QueryError::NoResponse(
+                    err.to_string(),
+                ))
+            })?
+            .data;
+        println!("Dry-run result: {}", result);
+        Ok(result)
+    }
+
+    /// Lookup the full response accompanying the specified transaction event
+    pub fn query_tx_response(
+        tendermint_addr: &str,
+        tx_query: TxEventQuery<'_>,
```

Comment:
This is essentially a hash, right? Can this type be instantiated by the caller?

Reply:
yeah, I changed this to take a `&str`. It means you can only query applied txs and not accepted, but I think that is totally fine.